features/server/electricalconnection.go

```diff
@@ -31,6 +31,48 @@ func NewElectricalConnection(localEntity spineapi.EntityLocalInterface) (*Electr
 	return ec, nil
 }
 
+func (e *ElectricalConnection) GetOrAddIdForDescription(
+	electricalConnectionDescription model.ElectricalConnectionDescriptionDataType,
+) (*model.ElectricalConnectionIdType, error) {
+	electricalConnectionId := (*model.ElectricalConnectionIdType)(nil)
+	highestExistingElectricalConnectionId := model.ElectricalConnectionIdType(0)
+
+	descriptionData := e.featureLocal.DataCopy(model.FunctionTypeElectricalConnectionDescriptionListData).(*model.ElectricalConnectionDescriptionListDataType)
+
+	if descriptionData != nil && descriptionData.ElectricalConnectionDescriptionData != nil {
+		for _, description := range descriptionData.ElectricalConnectionDescriptionData {
+			if description.ElectricalConnectionId != nil &&
+				description.PowerSupplyType == electricalConnectionDescription.PowerSupplyType &&
+				description.AcConnectedPhases == electricalConnectionDescription.AcConnectedPhases &&
+				description.AcRmsPeriodDuration == electricalConnectionDescription.AcRmsPeriodDuration &&
+				description.PositiveEnergyDirection == electricalConnectionDescription.PositiveEnergyDirection &&
+				description.ScopeType == electricalConnectionDescription.ScopeType &&
+				description.Label == electricalConnectionDescription.Label &&
+				description.Description == electricalConnectionDescription.Description {
+				electricalConnectionId = description.ElectricalConnectionId
+				break
```

Comment:
How about returning here directly? That would remove the need for the if statement in line 62